### PR TITLE
Make 'dab group start' fail fast

### LINKED ISSUE #418: dab group run doesn't fail fast
### Describe the bug

The `dab group run $group` command doesn't fail fast. The `dab group run $group` command runs individual `dab repo entrypoint run` commands. If one `dab repo entrypoint run` command in the group fails the subsequent commands continue rather than failing and displaying a message.

This behaviour regressed in this change:
https://github.com/Nekroze/dab/pull/409/files

It makes it rather hard to troubleshoot issues if starting your group provides a lot of output and the error disappears from screen. Failing fast would allow you to see the source of the error immediately and rectify.

### PR DESCRIPTION
**Pull request by tommymcguiver**
## Change Description

Make 'dab group start' fail fast

## Relevant Issue(s)

- Closes and fixes #418

### CODE REVIEW

File: app/subcommands/group/run [hunk 1]
```diff
@@ -12,7 +12,10 @@ run_with_dependencies() {
 	configkey="$2"
 	configpath="$(config_path "$configkey")"
 	[ -f "$configpath" ] || return 0
-	xargs -I{} sh -c "$depcmd {}" <"$configpath"
+	while read -r dep; do
+		# shellcheck disable=SC2086
+		$depcmd $dep </dev/null
+	done <"$configpath"
```
Comment (Nekroze) [inline app/subcommands/group/run]: This does not work for passing arguments in as read and while will cause it to split on all whitespace, not just new lines.

File: tests/features/group.feature [hunk 2]
```diff
@@ -41,6 +41,40 @@ Feature: Subcommand: dab group
 		Executing four entrypoint start
 		"""
 
+	Scenario: Can group repositories fail fast
+		Given I successfully run `dab repo add successOne https://github.com/Nekroze/dotfiles.git`
+		And I run `dab repo entrypoint create successOne start`
+		Given I successfully run `dab repo add failure https://github.com/Nekroze/dotfiles.git`
```
Comment (Nekroze) [inline tests/features/group.feature]: I think the linter will want this block to only have one given iirc.

Comment (tommymcguiver) [reply tests/features/group.feature]: Yea linter didn't barf but i'll update it to be consistent.

File: app/subcommands/group/run (hunk 1 above)
Comment (tommymcguiver) [reply app/subcommands/group/run]: Seems to work ok https://github.com/Nekroze/dab/blob/master/tests/features/group.feature#L72 passes

Comment (Nekroze) [reply app/subcommands/group/run]: huh, then I have no idea why I switched to xargs haha. Too long ago.

Comment (Nekroze) [reply app/subcommands/group/run]: lol wasn't even me. Love your work guys!

File: tests/features/group.feature [hunk 3]
```diff
@@ -22,6 +22,10 @@ Feature: Subcommand: dab group
 	Scenario: Can group repositories then start them together
 		Given I successfully run `dab repo add three https://github.com/Nekroze/dotfiles.git`
 		And I run `dab repo entrypoint create three start`
+		And I append to "~/.config/dab/repo/three/entrypoint/start" with:
+		"""
+		cat
```
Comment (Nekroze) [inline tests/features/group.feature]: won't this cause it to just block until stdin is closed? not sure if stdin is even available during run without piping.

File: app/subcommands/group/run [hunk 4]
```diff
@@ -12,7 +12,10 @@ run_with_dependencies() {
 	configkey="$2"
 	configpath="$(config_path "$configkey")"
 	[ -f "$configpath" ] || return 0
-	xargs -I{} sh -c "$depcmd {}" <"$configpath"
+	while read -r dep; do
+		# shellcheck disable=SC2086
+		$depcmd $dep </dev/null || fatality "$depcmd $dep exploded"
```
Comment (Nekroze) [inline app/subcommands/group/run]: why pipe `/dev/null` into stdin for the depcmd?

Comment (tommymcguiver) [reply app/subcommands/group/run]: It stops $depcmd from reading the rest of the input from stdin from parent process.

**Pull request merged**